src/framework/mpas_stream_manager.F

```diff
@@ -4845,6 +4964,51 @@ logical function is_decomposed_dim(dimName) !{{{
 
     end function is_decomposed_dim !}}}
 
+    
+    
+    integer function global_dim_size(block, dimNames, dimSizes, nDim) !{{{
+    use iso_c_binding, only : c_sizeof
+
+    implicit none
+
+    character(len=*), intent(in), dimension(:) :: dimNames
+    integer, intent(in), dimension(:) :: dimSizes
+    type(block_type), intent(in) :: block
+    integer, intent(in) :: nDim
+    integer, pointer :: block_dim_size
+    integer :: iDim, sum_block_dim_size
+    character(len=StrKIND):: message
+
+    global_dim_size = 1
+    do iDim = 1, nDim
+        if ( is_decomposed_dim(dimNames(iDim))) then
+            if (trim(dimNames(iDim)) == 'nCells') then 
+                call mpas_pool_get_dimension(block % dimensions, 'nCellsSolve', block_dim_size)
+            else if (trim(dimNames(iDim)) == 'nEdges') then 
+                call mpas_pool_get_dimension(block % dimensions, 'nEdgesSolve', block_dim_size)
+            else if  (trim(dimNames(iDim)) == 'nVertices') then 
+                call mpas_pool_get_dimension(block % dimensions, 'nVerticesSolve', block_dim_size)
+            else
+                global_dim_size = -1
+            end if
+            
+            call mpas_dmpar_sum_int(block % domain % dminfo, block_dim_size, sum_block_dim_size)
+            call mpas_log_write('----- from global_dim_size ----- Dimname is decomposed '//trim(dimNames(iDim))//' local size $i global size $i ',intArgs=(/block_dim_size, sum_block_dim_size/))
+            
+            global_dim_size = global_dim_size * sum_block_dim_size
+        else
+            global_dim_size = global_dim_size * dimSizes(iDim)
```

Comment:
Rather than using the `dimSizes` array for non-decomposed dimensions and querying dimension sizes for decomposed dimensions, it might simplify this function (and its argument list) to just query the dimension sizes in all cases with `mpas_pool_get_dimension`.

Reply:
Done. It now skips over the first dimension, if `isVarArray` is true